lib/topological_inventory/amazon/collector.rb

```diff
@@ -114,6 +122,9 @@ def save_inventory(collections, refresh_state_uuid = nil, refresh_state_part_uui
             :refresh_state_part_uuid => refresh_state_part_uuid,
           )
         )
+      rescue => e
+        logger.error("Error when sending payload to Ingress API. Error message: #{e.message}. Error headers: #{e&.response_headers}")
```

Comment:
`e&.response_headers`

The save navigator doesn't handle method_missing:
```
>> begin; raise; rescue => e; puts e&.response_headers; end
Traceback (most recent call last):
        3: from /usr/bin/irb:11:in `<main>'
        2: from (irb):6
        1: from (irb):6:in `rescue in irb_binding'
NoMethodError (undefined method `response_headers' for RuntimeError:RuntimeError)
```

Reply:
ok, so I should limit this on the API error only

Reply:
Yeah or either `.try()` or `if e.respond_to?()` would work